### DISCUSSION TITLE
What do you think about maplibre ?

### DISCUSSION THREAD

**Discussion by jesussmile** [liodali/osm_flutter#569] [General]
I tried all kind of flutter map for raster and everything has a performance issue. Then I tried maplibre for mapping. It is really fast and the performance is really good. Supports both raster and vector and many others features.  Will we see something like this ? Is it possible to integrate maplibre ? They have maplibre native for ios and android.

**Comment by liodali** [comment]
maplibre one of library that in my list to use for vector tiles for android sine osmdroid doesnt support vector tiles
or there (mapsforge/vtm)[https://github.com/mapsforge/vtm] not bad and this sdk not bad (mapscore)[https://github.com/openmobilemaps/maps-core] I built osm wrapper which I use here for ios side
but yeah maplibre most solid for vector tiles

**Reply by jesussmile** [reply]
What about ios will it not use vector tiles ?

**Reply by liodali** [reply]
it will but the implementation will much easier that android since osmdroid dont support vector tile 
i started making the implementation for android
hope be shipped too soon at the end of this month hopefully

**Reply by jesussmile** [reply]
Good luck. Hopefully the performance is good.

**Reply by liodali** [reply]
im redesigning the inner impl which will make it easy to maintain and ever for future changes will not take long time.
the api of maplibre not like osmdroid.
i will not say difficult but for someone get used to osmdroid i need some hours to get used to maplibre api.
i will start testing soon
and we will  have some break changes since i will remove  calling the osrm for each platform and start usage of our api client using dart that will make it easier even for drawing lines or even removed 😅 from the plugin and devs will have freedoom to use any routing api

**Reply by jesussmile** [reply]
Have you seen this package?  https://pub.dev/packages/maplibre

**Reply by liodali** [reply]
i saw it now 
nice package could be a goof alternative for osm_plugin
